### DISCUSSION TITLE
How to set direction value?

### DISCUSSION THREAD

**Discussion by TicTak21** [A-l-y-l-e/Alyle-UI#181] [Q&A] [answered]
I want to set direction(ltr/rlt) value. 
Is it possible? Something like `theme.setDirection(Dir.rtl)`

**Comment by Enlcxx** [comment, marked as answer]
You can use `LyTheme2` service

```ts
import { ..., LyTheme2 } from '@alyle/ui';

export class MyComponent {
  constructor(
    ...
    private _theme: LyTheme2,
  ) {}
  toggleDirection() {
    this._theme.toggleDirection()
  }
}
```
But to set direction (rtl/ltr) value it is not possible.
This method can be added, but I don't think it's useful, is it?

**Reply by TicTak21** [reply]
I think it could be useful because:

1) Consistent API, same as `theme.setTheme(themeName)`. Even for two values(ltr/rtl)
2) I'v got the case where I need to set direction from localstorage:

```js
const dir = localstorage.dir; // rtl
// after refresh page will be rolled back to ltr
// so I need to write some kind of check to sync page with localstorage value
if (theme.ltr) theme.toggleDirection()
```

While using setter I could just set value without any checks:

```js
const dir = localstorage.dir;
theme.setDir(dir);
```

This is the rare case, so I am okay with "toggle" solution.
Thank you.
